impl/src/test/groovy/com/okta/sdk/impl/oauth2/AccessTokenRetrieverServiceImplTest.groovy

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2020-Present Okta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.okta.sdk.impl.oauth2
+
+import com.okta.sdk.client.Client
+import com.okta.sdk.impl.Util
+import com.okta.sdk.impl.config.ClientConfiguration
+import io.jsonwebtoken.Claims
+import io.jsonwebtoken.Jwts
+import org.testng.annotations.Test
+
+import java.security.KeyPair
+import java.security.KeyPairGenerator
+import java.security.PrivateKey
+
+import static org.testng.Assert.assertEquals
+import static org.testng.Assert.assertNotNull
+
+import static org.mockito.Mockito.*
+
+/**
+ *
+ * Test for {@link AccessTokenRetrieverServiceImpl} class
+ */
+class AccessTokenRetrieverServiceImplTest {
+
+    @Test
+    void testInstantiationWithNullClientConfig() {
+        Util.expect(IllegalArgumentException) {
+            new AccessTokenRetrieverServiceImpl(null)
+        }
+    }
+
+    @Test
+    void testGetPrivateKeyFromPem() {
+
+        PrivateKey generatedPrivateKey = generatePrivateKey("RSA", 2048)
+        File privateKeyPemFile = writePrivateKeyToPemFile(generatedPrivateKey, "privateKey")
+
+        // Now test the pem -> private key conversion function of getPrivateKeyFromPem method
+        Reader pemFileReader = new FileReader(privateKeyPemFile)
+
+        PrivateKey resultPrivateKey = getAccessTokenRetrieverServiceInstance().getPrivateKeyFromPEM(pemFileReader)
+
+        privateKeyPemFile.deleteOnExit()
+
+        assertNotNull(resultPrivateKey, "privateKey cannot be null")
```

Comment:
nit: hamcrest will include these types of messages for you, using something like:

`assertThat(resultPrivateKey, notNullValue())`

`assertThat(resultPrivateKey.getAlgorithm(), is("RSA")`

Reply:
Fixed.